gcloud/storage/blob.py

```diff
@@ -287,6 +287,9 @@ def download_to_file(self, file_obj, client=None):
         :raises: :class:`gcloud.exceptions.NotFound`
         """
         client = self._require_client(client)
+        if self.media_link is None:  # not yet loaded
+            self.reload()
```

Comment:
I'm not seeing anything in error or missing in that docstring:  do you want to call out the fact that we might make an extra API call?  That feels to me like an implementation detail.